internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -131,6 +135,101 @@ func deleteEventById(conn redis.Conn, id string) (edgeXerr errors.EdgeX) {
 	return edgeXerr
 }
 
+// AsyncDeleteEvents Handles the deletion of device readings asynchronously. This function is expected to be running
+// in a go-routine and works with the "deleteReadingsByIds" function for better performance.
+func (c *Client) AsyncDeleteEvents() {
+	c.loggingClient.Debug("Starting background event deletion process")
+	for {
+		select {
+		case eventIds, ok := <-deleteEventsByIdChannel:
+			if ok {
+				c.loggingClient.Debug(fmt.Sprintf("Prepare to delete %v reading data", len(eventIds)))
+				startTime := time.Now()
+				conn := c.Pool.Get()
+				err := deleteEventsByIds(conn, eventIds, c.BatchSize)
+				if err != nil {
+					c.loggingClient.Error(fmt.Sprintf("Deleted events failed.  Err: %s", err.Error()))
+				} else {
+					c.loggingClient.Debug(fmt.Sprintf("Deleted events successfully. elapsed time: %s", time.Since(startTime)))
+				}
+				conn.Close()
+			}
+		}
+	}
+}
+
+// deleteEventsByIds deletes all readings with given event Ids
```

Comment:
This seems to only delete the events, not readings.

Reply:
comments updated.